orchestrator/src/main/kotlin/Orchestrator.kt

```diff
@@ -340,9 +332,22 @@ class Orchestrator(
 
         val ortRun = getCurrentOrtRun(ortRunId)
         val scheduleContext = createWorkerSchedulerContext(ortRun, header, workerJobs = jobs)
-        val schedules = WorkerScheduleInfo.entries.mapNotNull { it.createAndScheduleJobIfPossible(scheduleContext) }
 
-        return scheduleContext to schedules
+        return fetchNextJobs(scheduleContext)
+    }
+
+    /**
+     * Convenience function to evaluate and process this [Result] with information about the next jobs to be scheduled.
+     * If the result is successful, actually trigger the jobs. Otherwise, call the given [onError] function with the
+     * exception that occurred.
+     */
+    private fun Result<Pair<WorkerScheduleContext, List<JobScheduleFunc>>>.scheduleNextJobs(
+        onError: (Throwable) -> Unit
```

Comment:
Don't you want to stick with `onFailure` ? Introducing the new name `onError` is unecessary complex IMO